File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)
        ans = 0
        for i in nums:
            if i-1 not in seen:
                length = 1

                while i+length in seen:
                    length+=1
                ans = max(ans,length)
        return ans

        

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int solve(TreeNode* root, int m){
        if(!root) return 0;
        int ans=0;
        int max = m;
        if(root->val>=m){
            ans = 1;
            max=root->val;
        }
        return ans+solve(root->left,max)+solve(root->right,max);
    }
    int goodNodes(TreeNode* root) {
        return solve(root,INT_MIN);
    }
};
